test/functional/services/filter_bar.js

```diff
@@ -44,18 +45,24 @@ export function FilterBarProvider({ getService }) {
       const filterElement = await testSubjects.find(`filter & filter-key-${key}`);
       await remote.moveMouseTo(filterElement);
       await testSubjects.click(`filter & filter-key-${key} removeFilter-${key}`);
+      // these methods should each wait until their action is complete but we don't know what to wait for
+      await PageObjects.common.sleep(2001);
     }
 
     async toggleFilterEnabled(key) {
       const filterElement = await testSubjects.find(`filter & filter-key-${key}`);
       await remote.moveMouseTo(filterElement);
       await testSubjects.click(`filter & filter-key-${key} disableFilter-${key}`);
+      // these methods should each wait until their action is complete but we don't know what to wait for
+      await PageObjects.common.sleep(2002);
```

Comment:
The filter element should have a data-test-subj that goes from `filter-enabled` to `filter-disabled` and the global loading indicator should fire, can we look for those?

Reply:
I'm doubting whether checking the state of the filter pill is a good indication that the rest of the changes in the UI are complete.  It seems like that filter change would be the first thing that happens and everything else would still be changing?

The other issue here is that this method toggles, so to wait for that change I'd need to first get that state and then make the wait depend on the opposite state.  Not too hard, but not sure it's worth it (see above).

Or, I could change it so we have `enableFilter` and `disableFilter` methods.  Then the check that it has changed is simple.  I think that would simplify the test too.  Right now when you read the test and it says toggleFilterEnabled you don't know if it's enabling or disabling the filter.
What do you think ^?

Reply:
> I'm doubting whether checking the state of the filter pill is a good indication that the rest of the changes in the UI are complete. It seems like that filter change would be the first thing that happens and everything else would still be changing?

Waiting just for the filter change definitely isn't sufficient, that's why I mentioned the global loading indicator as well.

> I could change it so we have enableFilter and disableFilter methods. Then the check that it has changed is simple.

I think that's probably a good idea.

Reply:
For now I've just added this and the tests have passed several times on Jenkins.
`await PageObjects.header.awaitGlobalLoadingIndicatorHidden();`